pkg/ddl/backfilling_dist_executor.go

```diff
@@ -145,9 +150,31 @@ func (s *backfillDistExecutor) newBackfillStepExecutor(
 	jobMeta := &s.taskMeta.Job
 	ddlObj := s.d
 
+	store := ddlObj.store
+	sessPool := ddlObj.sessPool
+	taskKS := s.task.Keyspace
+	if keyspace.IsRunningOnSystem() && taskKS != config.GetGlobalKeyspaceName() {
```

Comment:
```suggestion
	if taskKS != config.GetGlobalKeyspaceName() {
```
`taskKS != config.GetGlobalKeyspaceName()` implies it's running on SYSTEM

Reply:
I think it is better to make it explicitly to avoid unexpected executions. (Make `kernel.NextGen()` and `SYSTEM` explicitly)

Reply:
maybe add a comment about why we still state it explicitly even though `taskKS != config.GetGlobalKeyspaceName() implies it's running on SYSTEM`

Reply:
If you want to avoid unexpected executions, it's better to use intest.Assert to let it panic in test environment